tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1482 @@
+    val resp = getWhiskEntity(path).futureValue
+    var r = new RestResult(resp.status, getRespData(resp))
+    if (expectedExitCode != DONTCARE_EXIT) {
+      r.statusCode.intValue shouldBe expectedExitCode
+    }
+    r
+  }
+
+  /**
+   * Gets activation result by id.
+   *
+   * @param activationId the activation id
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  def activationResult(activationId: String, expectedExitCode: Int = OK.intValue)(implicit wp: WskProps): RestResult = {
+    val path = Path(s"${basePath}/namespaces/${wp.namespace}/$noun/$activationId/result")
+    val resp = getWhiskEntity(path).futureValue
+    var r = new RestResult(resp.status, getRespData(resp))
+    if (expectedExitCode != DONTCARE_EXIT) {
+      r.statusCode.intValue shouldBe expectedExitCode
+    }
+    r
+  }
+
+  /**
+   * Polls activations list for at least N activations. The activations
+   * are optionally filtered for the given entity. Will return as soon as
+   * N activations are found. If after retry budget is exhausted, N activations
+   * are still not present, will return a partial result. Hence caller must
+   * check length of the result and not assume it is >= N.
+   *
+   * @param N the number of activations desired
+   * @param entity the name of the entity to filter from activation list
+   * @param limit the maximum number of entities to list (if entity name is not unique use Some(0))
+   * @param since (optional) only the activations since this timestamp are included
+   * @param retries the maximum retries (total timeout is retries + 1 seconds)
+   * @return activation ids found, caller must check length of sequence
+   */
+  override def pollFor(N: Int,
+                       entity: Option[String],
+                       limit: Option[Int] = Some(30),
+                       since: Option[Instant] = None,
+                       retries: Int = 10,
+                       pollPeriod: Duration = 1.second)(implicit wp: WskProps): Seq[String] = {
+    Try {
+      retry({
+        val result = idsActivation(listActivation(filter = entity, limit = limit, since = since, docs = false))
+        if (result.length >= N) result else throw PartialResult(result)
+      }, retries, waitBeforeRetry = Some(pollPeriod))
+    } match {
+      case Success(ids)                => ids
+      case Failure(PartialResult(ids)) => ids
+      case _                           => Seq()
+    }
+  }
+
+  override def get(activationId: Option[String],
+                   expectedExitCode: Int = OK.intValue,
+                   fieldFilter: Option[String] = None,
+                   last: Option[Boolean] = None)(implicit wp: WskProps): RestResult = {
+    val path = activationId map { id =>
+      Path(s"${basePath}/namespaces/${wp.namespace}/$noun/$id")
+    } getOrElse Path("")
```

Comment:
What happens if path is an empty string?

Reply:
We should return a not found error.